### DISCUSSION TITLE
Database pricing questions

### DISCUSSION THREAD

**Discussion by testdgb** [supabase/supabase#13864] [Questions]
Hey there, I'm interested in using supabase for my backend and database cloud server but I have a few questions to ask about the pricing - mainly for the egress pricing. (I'm developing with next.js and node)

Database egress price only counts for READS operations (e.g supabase.from('users')...)? Does WRITE operations don't count for my billing at all? If I make an API read calls for a file size of 50kb, then I make some changes in it inside the TSX and update it into the serves (same size). Will I be paid for 50kb of egress or 100kb?

If I store the user's all images public link in an array cell inside the users table, It will save me the Storage egress for each time of retrieving the images. right?

Is there a way to retrieve just half of JSONB file using node request, to save the egress usage? 
Thank you!

**Comment by GaryAustin1** [comment]
Only data leaving the server API is counted.  The docs are not clear if that includes http overhead.  Assuming it does inserts and updates without returning data would have a small amount of egress for the status response. 

Not sure what 1/2 of JSON data means.  You can select parts of the JSON with json col->jsonsubpart and get only the json from that point on.
